appender/loki/src/main/java/org/apache/karaf/decanter/appender/loki/LokiAppender.java

```diff
@@ -0,0 +1,116 @@
+ * limitations under the License.
+ */
+package org.apache.karaf.decanter.appender.loki;
+
+import org.apache.karaf.decanter.api.marshaller.Marshaller;
+import org.apache.karaf.decanter.appender.utils.EventFilter;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventConstants;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Dictionary;
+
+@Component(
+        name = "org.apache.karaf.decanter.appender.loki",
+        immediate = true,
+        property = EventConstants.EVENT_TOPIC + "=decanter/collect/*"
+)
+public class LokiAppender implements EventHandler {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(LokiAppender.class);
+
+    @Reference(cardinality = ReferenceCardinality.OPTIONAL)
+    public Marshaller marshaller;
+
+    private String url;
+    private String tenant = null;
+    private String username = null;
+    private String password = null;
+    private Dictionary<String, Object> config;
+
+    @Activate
+    public void activate(ComponentContext componentContext) {
+        activate(componentContext.getProperties());
+    }
+
+    public void activate(Dictionary<String, Object> config) {
+        this.config = config;
+        url = (config.get("loki.url") != null) ? (String) config.get("loki.url") : "http://localhost:3100/loki/api/v1/push";
+        tenant = (config.get("loki.tenant") != null) ? (String) config.get("loki.tenant") : null;
+        username = (config.get("loki.username") != null) ? (String) config.get("loki.username") : null;
+        password = (config.get("loki.password") != null) ? (String) config.get("loki.password") : null;
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        if (EventFilter.match(event, config)) {
+            String log;
+            if (marshaller != null) {
+                log = marshaller.marshal(event);
+            } else {
+                StringBuilder builder = new StringBuilder();
+                for (String innerKey : event.getPropertyNames()) {
+                    builder.append(innerKey).append(":").append(toString(event.getProperty(innerKey))).append(" | ");
+                }
+                log = builder.toString();
+            }
+            try {
+                HttpURLConnection connection = (HttpURLConnection) new URL("http://localhost:3100/loki/api/v1/push").openConnection();
```

Comment:
@jbonofre we should use `url` private attribute here instead off hard code url, right?

Reply:
Good catch ! Thanks for spotting this !